lib/std/jsbigints.nim

```diff
@@ -0,0 +1,186 @@
+  runnableExamples:
+    doAssert (newJsBigInt"999" shl newJsBigInt"2") == newJsBigInt"3996"
+
+func `shr`*(a, b: JsBigInt): JsBigInt {.importjs: "(# >> #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"999" shr newJsBigInt"2") == newJsBigInt"249"
+
+func `-`*(this: JsBigInt): JsBigInt {.importjs: "($1#)".} =
+  runnableExamples:
+    doAssert -(newJsBigInt"10101010101") == newJsBigInt"-10101010101"
+
+func inc*(this: var JsBigInt) {.importjs: "(++[#][0][0])".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"1"
+    inc big1
+    doAssert big1 == newJsBigInt"2"
+
+func dec*(this: var JsBigInt) {.importjs: "(--[#][0][0])".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"2"
+    dec big1
+    doAssert big1 == newJsBigInt"1"
+
+func inc*(this: var JsBigInt; amount: JsBigInt) {.importjs: "([#][0][0] += #)".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"1"
+    inc big1, newJsBigInt"2"
+    doAssert big1 == newJsBigInt"3"
+
+func dec*(this: var JsBigInt; amount: JsBigInt) {.importjs: "([#][0][0] -= #)".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"1"
+    dec big1, newJsBigInt"2"
+    doAssert big1 == newJsBigInt"-1"
+
+func `+=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"1"
+    big1 += newJsBigInt"2"
+    doAssert big1 == newJsBigInt"3"
+
+func `-=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"1"
+    big1 -= newJsBigInt"2"
+    doAssert big1 == newJsBigInt"-1"
+
+func `*=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"2"
+    big1 *= newJsBigInt"4"
+    doAssert big1 == newJsBigInt"8"
+
+func `/=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] /= #)".} =
+  ## Same as `x = x div y`.
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"11"
+    big1 /= newJsBigInt"2"
+    doAssert big1 == newJsBigInt"5"
+
+proc `+`*(_: JsBigInt): JsBigInt {.error:
+  "See https://github.com/tc39/proposal-bigint/blob/master/ADVANCED.md#dont-break-asmjs".} # Can not be used by design
+  ## **Do NOT use.** https://github.com/tc39/proposal-bigint/blob/master/ADVANCED.md#dont-break-asmjs
+
+proc low*(_: typedesc[JsBigInt]): JsBigInt {.error:
+  "Arbitrary precision integers do not have a known low.".} ## **Do NOT use.**
+
+proc high*(_: typedesc[JsBigInt]): JsBigInt {.error:
+  "Arbitrary precision integers do not have a known high.".} ## **Do NOT use.**
+
+
+runnableExamples:
+  let big1: JsBigInt = newJsBigInt"2147483647"
+  let big2: JsBigInt = newJsBigInt"666"
+  doAssert JsBigInt isnot int
+  doAssert big1 != big2
+  doAssert big1 > big2
+  doAssert big1 >= big2
+  doAssert big2 < big1
+  doAssert big2 <= big1
+  doAssert not(big1 == big2)
+  let z = JsBigInt.default
+  doAssert $z == "0"
```

Comment:
```suggestion
  doAssert $z == "0"
  block:
    var a: seq[JsBigInt]
    a.setLen 2
    doAssert a == @[newJsBigInt"0", newJsBigInt"0"]
    doAssert a[^1] == newJsBigInt"0"
    var b: JsBigInt
    doAssert b == newJsBigInt"0"
    doAssert b == JsBigInt.default
```

(to illustrate value semantics; this wouldn't pass before the change involving `{.importc: "bigint".}`)